### ISSUE TITLE
pad checking failed.

### ISSUE THREAD

**Issue by a1ad** [mcdope/pam_usb#96]
```
[Jeroen@400833016 ~]$ pamusb-check Jeroen --debug
* Authentication request for user "Jeroen" (pamusb-check)
[src/local.c:212] Checking whether the caller (pamusb-check) is local or not...
[src/local.c:222] 	Checking pid  36714 (pamusb-check)...
[src/local.c:222] 	Checking pid  35898 (bash)...
[src/local.c:222] 	Checking pid  28230 (/usr/libexec/gnome-terminal-server)...
[src/local.c:222] 	Checking pid   2252 (/usr/lib/systemd/systemd)...
[src/local.c:222] 	Checking pid      1 (/usr/lib/systemd/systemd)...
[src/local.c:271] 	Using DISPLAY :0 for utmp search
[src/local.c:047] 	No utmp entry found for tty ":0"
[src/local.c:284] 	Trying to get tty from display server
[src/local.c:289] 	Retrying with tty /dev/tty2, obtained from display server, for utmp search
[src/local.c:050] 		utmp entry for tty "tty2" found
[src/local.c:051] 			utmp->ut_pid: 23691
[src/local.c:052] 			utmp->ut_user: Jeroen
[src/local.c:066] 		Checking utmp->ut_addr_v6[0]
[src/local.c:066] 		Checking utmp->ut_addr_v6[1]
[src/local.c:066] 		Checking utmp->ut_addr_v6[2]
[src/local.c:066] 		Checking utmp->ut_addr_v6[3]
[src/local.c:070] 	utmp check successful, request originates from a local source!
[src/local.c:338] No remote access detected, seems to be local request - allowing.
[src/device.c:038] Searching for "JeroKEY" in the hardware database...
* Authentication device "JeroKEY" is connected.
* Performing one time pad verification...
[src/volume.c:097] Searching for volume with uuid D0CD-159C.
[src/volume.c:121] Found mount points: /run/media/Jeroen/D0CD-159C
[src/volume.c:134] Found volume D0CD-159C.
[src/volume.c:158] Volume D0CD-159C is already mounted.
[src/pad.c:061] Cannot open device file: No such file or directory
* Pad checking failed!
* Access denied.
```

What did I do wrong??

Version: 
git status
```
On branch issue-11-provide-optional-unlock-script
Your branch is up to date with 'origin/issue-11-provide-optional-unlock-script'.
```

**Comment by mcdope** [comment]
This happens if the pad on the device and your computer become out of sync. i.e this happens often in testing for me when i use the same device on multiple computers/VMs.

I don't remember the exact fix, it was either deleting the pads in the homedir or on the device.

**Comment by a1ad** [comment]
Can't this be fixed automatically? When pads checking failed > remove pads and create new ones?

**Comment by mcdope** [comment]
That would defeat the purpose of the pads totally. Just disable them if you want that.

In normal use this doesn't happen.

**Comment by a1ad** [comment]
aha, that's it. I disconnected without proper dismount. Thanks for the clarification!

**Comment by mcdope** [comment]
Nice to know how it happened. 

Will check if I can somehow avoid that but guess it's related to buffering stuff on the device since we flush() our buffers already after pad updates. But will investigate.

**Comment by a1ad** [comment]
Or cant you leave a hint? like: Remove pads folder at this and this location to reset? Or Just a confirmation message: like do you want me to reset the pads folders "your device is possible tampered". Now its just a guessing game for the enduser.

**Comment by mcdope** [comment]
Yeah, would be good to add. But nonetheless I would prefer to avoid that behavior altogether

**Comment by mcdope** [comment]
In your specific case it was seemingly caused by the removal of the auth device while being mounted. So the system thought the media is still there, but of course the directories/files weren't accessible anymore. 

For this specific case I've added a info level output hinting to it as the culprit. For other cases, which I've falsely assumed to be the case here initially, I've added a paragraph into the docs at https://github.com/mcdope/pam_usb/wiki/Troubleshooting#getting-pad-checking-failed-when-trying-to-authenticate.

I consider this done for now.
